### PR TITLE
feat(trie): make benchmark inputs deterministic

### PR DESCRIPTION
**Pull request by hoank101**

### CODE REVIEW

File: crates/trie/common/benches/prefix_set.rs [hunk 1]
```diff
@@ -115,8 +111,7 @@ fn prefix_set_bench<T>(
 fn generate_test_data(size: usize) -> (Vec<Nibbles>, Vec<Nibbles>, Vec<bool>) {
     use prop::collection::vec;
 
-    let config = ProptestConfig { result_cache: basic_result_cache, ..Default::default() };
-    let mut runner = TestRunner::new(config);
+    let mut runner = TestRunner::deterministic();
```
Comment (Rjected) [inline crates/trie/common/benches/prefix_set.rs:114]: I'm not sure what impact the result cache originally had on benchmark performance here, but can we keep it? See the proptest docs for what the deterministic function desugars to:
https://docs.rs/proptest/latest/proptest/test_runner/struct.TestRunner.html#method.deterministic

**Review by Rjected — CHANGES_REQUESTED** [review]
just one nit

File: crates/trie/common/benches/prefix_set.rs (hunk 1 above)
Comment (hoank101) [reply crates/trie/common/benches/prefix_set.rs]: thank you, i just refactor the code, please help me review

File: crates/trie/common/benches/prefix_set.rs [hunk 2]
```diff
@@ -115,8 +115,13 @@ fn prefix_set_bench<T>(
...
 
-    let config = ProptestConfig { result_cache: basic_result_cache, ..Default::default() };
-    let mut runner = TestRunner::new(config);
+    let config = ProptestConfig {
+        result_cache: basic_result_cache,
+        rng_algorithm: RngAlgorithm::ChaCha,
+        ..Default::default()
+    };
+    let mut runner =
+        TestRunner::new_with_rng(config, TestRng::deterministic_rng(RngAlgorithm::ChaCha));
```
Comment (Rjected) [inline crates/trie/common/benches/prefix_set.rs:124]: would be nice if we could do this
```suggestion
    let config = ProptestConfig {
        result_cache: basic_result_cache,
        ..Default::default()
    };
    let mut runner =
        TestRunner::new_with_rng(config, TestRng::deterministic_rng(config.rng_algorithm));
```

**Review by Rjected — CHANGES_REQUESTED** [review]
one very smol nit, otherwise this looks good

**Comment by hoank101** [comment]
> one very smol nit, otherwise this looks good

thank you, I just updated

**Review by Rjected — APPROVED** [review]
lgtm

**Comment by hoank101** [comment]
oh, i see duplicated

**Pull request closed without merging**